stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewHighlighter.java

```diff
@@ -0,0 +1,112 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements.android;
+
+import android.annotation.TargetApi;
+import android.graphics.drawable.ColorDrawable;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.View;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+abstract class ViewHighlighter {
+  public static ViewHighlighter newInstance() {
+    // TODO: find ways to do highlighting on older versions too
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return new OverlayHighlighter();
+    } else {
+      return new NullHighlighter();
```

Comment:
This represents a poor API.  If you are not going to provide a working implementation for all versions, the `newInstance` method should be marked `@Nullable` and you should actually return null.  This way callers explicitly have to write:
```java
if (mHighlighter != null) {
  mHighlighter.highlight(...)
}
```

An alternative strategy is to add an `isSupported` method that `NullHighlighter` (prefer `NoopHighlighter` as well since null has no semantic context here) implements and returns false, then throws on all other methods, forcing callers to write:
```java
if (mHighlighter.isSupported()) {
  mHighlighter.highlight(...);
}
```

The reason for being explicit is so that readers of the highlighting code can see plainly that it might not be supported and that would explain why it's not working for them on a naive/quick read of the source code.  Forcing folks to dig all the way down into the implementation to see that possibility is frustrating.